### PR TITLE
Added => Isomorphic.js

### PR DESCRIPTION
**Pull request by AnupamTekale**
## Assigned Issue Number ?
#1682 

## Location of the file(s) ?
Algorithms/String-Algorithm/Isomorphic.js

### Checklist

- [ YES] **I've followed the [contributing guidelines](https://github.com/ahampriyanshu/algo_ds_101/blob/main/CONTRIBUTING.md)**
- [ YES] The content added isn't directly copied from any sources like gfg, tutorialpoint, etc.
- [ NO] Input has been taken dynamically.
- [ YES] I've referred the correct issue number.
- [ YES] Comments have been added
- [ YES] I've  synced-up my forked repo.
- [ YES] Ain't pushing from the main branch.

### CODE REVIEW

**Review by ahampriyanshu — CHANGES_REQUESTED** [review]
Hi @AnupamTekale, please take the input dynamically.

**Comment by AnupamTekale** [comment]
@ahampriyanshu Okay!

File: Algorithms/String-Algorithm/Isomorphic.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+s = "abc"
+t = "aaa"
+
+var isIsomorphic = function(s, t) {
```
Comment (AnupamTekale) [inline Algorithms/String-Algorithm/Isomorphic.js:4]: var isomorphic = function(input[0],input[1])
{
let s= input[0]
let t = input[1]

}

File: Algorithms/String-Algorithm/Isomorphic.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            
+            if (m.get(s[i]) != t[i]) {
+                
+                return false
+            }
+        }
+    }
+    return new Set([...m.values()]).size == m.size
+    
+};
```
Comment (AnupamTekale) [inline Algorithms/String-Algorithm/Isomorphic.js:21]: //DYNAMICALLY TAKING STRINGS 
var input = []; // initialise an empty array
var temp = '';
do {
    temp = prompt("Enter a number. Press cancel or leave empty to finish.");
    if (temp === "" || temp === null) {
        break;
    } else {
        input.push(temp);  
    }
} while (1);

File: Algorithms/String-Algorithm/Isomorphic.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        else {
+            
+            if (m.get(s[i]) != t[i]) {
+                
+                return false
+            }
+        }
+    }
+    return new Set([...m.values()]).size == m.size
+
```
Comment (AnupamTekale) [inline Algorithms/String-Algorithm/Isomorphic.js:20]: /*We use a Map to record the key/val pair in S and T. For each char in S, if we never met it before, we record the (s[i], t[i]) pair in map. If we met s[i] before, we compare the value of s[i] with t[i], which are supposed to be the same. After that, we need to check whether map.values() contain duplicate value.*/

**Review by AnupamTekale — COMMENTED** [review]
I have done it dynamically @ahampriyanshu

**Review by ahampriyanshu — APPROVED** [review]
Perfect Code :+1: 
Thanks @AnupamTekale  
Happy Contributing

**Pull request merged**